roles/dd-nodepool/files/nodepool_server.py

```diff
@@ -55,3 +70,25 @@ def get_nodepool_status(self, status_url, timeout, service_check_tags):
                            AgentCheck.OK,
                            message='Received status from %s' % status_url,
                            tags=service_check_tags)
+
+    def get_nodepool_images(self, images_url, timeout, service_check_tags):
+        r = requests.get(images_url, timeout=timeout)
+        currentTime = int(time.time())
+        staleImage = False
+
+        for image in r.json():
+            if currentTime - image['age'] > 172800:
+                self.service_check(
+                    self.IMAGES_CHECK_NAME,
+                    AgentCheck.CRITICAL,
+                    message='Nodepool image is stale: %s' % image['id']
+                    tags=images_check_tags)
+                staleImage = True
+
+        if staleImage:
+            return
+
+        self.service_check(self.IMAGES_CHECK_NAME,
```

Comment:
I don't think you want this to be a service check. A service check is a good/bad check for when something has actually gone wrong. What you can do here is report to datadog the actual age of the image and then on the datadog side we say that it is something we need to be warned about if age > X. I think you probably want a gauge: http://docs.datadoghq.com/guides/dogstatsd/#metrics